source/common/http/conn_manager_utility.cc

```diff
@@ -93,6 +93,13 @@ Network::Address::InstanceConstSharedPtr ConnectionManagerUtility::mutateRequest
         connection.ssl() ? Headers::get().SchemeValues.Https : Headers::get().SchemeValues.Http);
   }
 
+  const auto& local_address = connection.localAddress();
+  // If x-forwarded-port is desired and remote hasn't set it (trusted proxy), we set it.
```

Comment:
I think we need to sanitize x-forwarded-port in non-trusted scenarios if this feature is enabled? See how/where we sanitize XFF.

Reply:
While I'm looking at it, should we also support multiple ports (and also protocols and probably hosts?). Ref: [zuul's pre decoration filter on x-forwarded-{port,proto} appends test case](https://github.com/spring-cloud/spring-cloud-netflix/blob/6e799423d5f2a95ae26895d9a3ac1eece762ab98/spring-cloud-netflix-zuul/src/test/java/org/springframework/cloud/netflix/zuul/filters/pre/PreDecorationFilterTests.java#L219-L220).

Reply:
We do support multiple XFF hosts. For protocols and ports, I don't have a strong opinion. I find it hard to believe that for protocols and ports people really care, but I'm not sure. I would keep it simple IMO for now and can always expand later, but up to you.